backend/src/s9pk/mod.rs

```diff
@@ -94,33 +99,73 @@ pub async fn pack(#[context] ctx: SdkContext, #[arg] path: Option<PathBuf>) -> R
                     )
                 })?,
         )
-        .docker_images(
-            File::open(path.join(manifest.assets.docker_images_path()))
-                .await
-                .with_ctx(|_| {
-                    (
-                        crate::ErrorKind::Filesystem,
-                        manifest.assets.docker_images_path().display().to_string(),
-                    )
-                })?,
-        )
+        .docker_images({
+            let docker_images_path = path.join(manifest.assets.docker_images_path());
```

Comment:
:thought_balloon: Wondering if this this be part of the struct of something, the logic being here feels like an abstraction fail.